### PR TITLE
Update e2e supported images test to support s390x arch.

### LINKED ISSUE #3670: Ask for images work on s390x arch
/kind bug

## What versions of software are you using?

**Operating System:** 
 Red Hat Enterprise Linux 8.1 (Ootpa

**Output of `odo version`:**
odo v1.2.4 (b3cf59550)
Server: https://api.ocp.openstack:6443
Kubernetes: v1.18.3+6025c28
## How did you run odo exactly?
Enable odo CI on s390x platform

## Actual behavior
./scripts/openshiftci-presubmit-all-tests.sh

## Expected behavior
All the test passed 

## Any logs, error output, etc?
There are some hard code to import S2I images. These images only support amd64 arch which causes failed on s390x arch. 
For odo supported images e2e tests:
These images are:
`oc --request-timeout 5m import-image nodejs:latest --namespace=vsnyjhnygd --from=docker.io/centos/nodejs-12-centos7:latest --confirm`
`oc --request-timeout 5m import-image nodejs:latest --namespace=hcvvpyounm --from=docker.io/centos/nodejs-10-centos7:latest --confirm`
`oc --request-timeout 5m import-image nodejs:latest --namespace=qjycpyjbmo --from=registry.access.redhat.com/rhoar-nodejs/nodejs-10:latest --confirm`
For odo source e2e tests:
`oc --request-timeout 5m import-image dotnet:2.0 --namespace=uezadyekmn --from=registry.centos.org/dotnet/dotnet-20-centos7 --confirm`

### PR DESCRIPTION
**Pull request by zhengxiaomei123**
**What type of PR is this?**
 /kind failing-test
**What does does this PR do / why we need it**:
There are some hard code to import S2I images. These images only support amd64 arch which causes failed on s390x arch.

**Which issue(s) this PR fixes**:

Fixes #3670 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: scripts/configure-installer-tests-cluster-s390x.sh [hunk 1]
```diff
@@ -70,9 +70,15 @@ for i in `echo $USERS`; do
     HTPASSWD_CREATED=""
 done
 
-# Workarounds - Note we should find better soulutions asap
-## Missing wildfly in OpenShift Adding it manually to cluster Please remove once wildfly is again visible
-#oc apply -n openshift -f https://raw.githubusercontent.com/openshift/library/master/arch/x86_64/community/wildfly/imagestreams/wildfly-centos7.json
+#Update the imagestream
```
Comment (jichenjc) [inline scripts/configure-installer-tests-cluster-s390x.sh:73]: should we add some comments here?

File: tests/e2escenarios/e2e_images_test.go [hunk 2]
```diff
@@ -85,15 +86,11 @@ var _ = Describe("odo supported images e2e tests", func() {
...
-	Context("odo supported images deployment", func() {
-		It("Should be able to verify the openjdk18-openshift image", func() {
-			oc.ImportImageFromRegistry("registry.access.redhat.com", filepath.Join("redhat-openjdk-18", "openjdk18-openshift:latest"), "java:8", project)
-			verifySupportedImage(filepath.Join("redhat-openjdk-18", "openjdk18-openshift:latest"), "openjdk", "java:8", project, appName, context)
-		})
-
-		It("Should be able to verify the openjdk-11-rhel7 image", func() {
-			oc.ImportImageFromRegistry("registry.access.redhat.com", filepath.Join("openjdk", "openjdk-11-rhel7:latest"), "java:8", project)
-			verifySupportedImage(filepath.Join("openjdk", "openjdk-11-rhel7:latest"), "openjdk", "java:8", project, appName, context)
+	Context("odo supported images deployment on x86", func() {
```
Comment (jichenjc) [inline tests/e2escenarios/e2e_images_test.go:89]: what about other platform? you are saying it's x86 ...

Comment (zhengxiaomei123) [reply tests/e2escenarios/e2e_images_test.go:89]: @jichenjc Those images are only available for x86. I don't replace it with ubi/nodejs but pick these up to the specific context. I think it a good solution to not mix x86 and s390x together.

File: scripts/configure-installer-tests-cluster-s390x.sh (hunk 1 above)
Comment (zhengxiaomei123) [reply scripts/configure-installer-tests-cluster-s390x.sh:73]: > should we add some comments here?

I will update with more info. Thanks.

File: tests/e2escenarios/e2e_images_test.go (hunk 2 above)
Comment (jichenjc) [reply tests/e2escenarios/e2e_images_test.go:89]: I am only talk about `power` :)

File: scripts/openshiftci-presubmit-all-tests.sh [hunk 3]
```diff
@@ -40,9 +40,7 @@ if [ "${ARCH}" == "s390x" ]; then
     make test-cmd-login-logout
     make test-cmd-project
     # E2e tests
-    make test-e2e-beta
-    make test-e2e-java
-    make test-e2e-source
+    make test-e2e-all
```
Comment (amitkrout) [inline scripts/openshiftci-presubmit-all-tests.sh:43]: +1

File: tests/e2escenarios/e2e_images_test.go (hunk 2 above)
Comment (zhengxiaomei123) [reply tests/e2escenarios/e2e_images_test.go:89]: > I am only talk about `power` :)
I got it. I will update the condition to `amd64` .

**Comment by girishramnani** [comment]
retesting to confirm that new flaky behaviour is not introduced.
/test all

**Comment by zhengxiaomei123** [comment]
> While we are at this can we also update the supported image matrix at https://github.com/openshift/odo#officially-supported-languages-and-corresponding-container-images, also indicating the archs on which the images are available?
> 
> cc @kadel wdyt?

Yes. It's our plan to update the documents to verify which images are available especially for 390x arch. I think all the updates of documents will request one PR. If necessary, I will update here.

File: tests/e2escenarios/e2e_images_test.go [hunk 4]
```diff
@@ -85,15 +86,11 @@ var _ = Describe("odo supported images e2e tests", func() {
...
+		JustBeforeEach(func() {
+			if runtime.GOARCH != "amd64" {
```
Comment (amitkrout) [inline tests/e2escenarios/e2e_images_test.go:91]: @zhengxiaomei123 As per the context it says supported image deployment on x86 architecture but in the condition it check for the opposite (runtime.GOARCH != "amd64" -> x86), i mean its skipping the test for same x86 architecture.

May be you need to update the context description like ```odo supported images deployment on amd64```

**Comment by amitkrout** [comment]
> > While we are at this can we also update the supported image matrix at https://github.com/openshift/odo#officially-supported-languages-and-corresponding-container-images, also indicating the archs on which the images are available?
> > cc @kadel wdyt?
> 
> @mohammedzee1000 Yes. It's our plan to update the documents to verify which images are available especially for 390x arch. I think all the updates of documents will request one PR. If necessary, I will update here.

@zhengxiaomei123 If you are not so sure about the doc atm then you can create an issue to track that else you can update the doc here too.

**Comment by zhengxiaomei123** [comment]
> If you are not so sure about the doc atm then you can create an issue to track that else you can update the doc here too.

About the images doc, I will update here.

File: tests/e2escenarios/e2e_images_test.go (hunk 4 above)
Comment (zhengxiaomei123) [reply tests/e2escenarios/e2e_images_test.go:91]: > @zhengxiaomei123 As per the context it says supported image deployment on x86 architecture but in the condition it check for the opposite (runtime.GOARCH != "amd64" -> x86), i mean its skipping the test for same x86 architecture.
> 
> May be you need to update the context description like `odo supported images deployment on amd64`

Sure. I will update it. I confused those two arch.

**Comment by zhengxiaomei123** [comment]
> Indentation somehow went wrong

@amitkrout Thanks for help verify it. How did you verify? please let me know. Thanks.

**Pull request merged**